src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -74,6 +75,47 @@ UT_STATIC void sys_comp_module_##adapter##_init(void) \
 \
 DECLARE_MODULE(sys_comp_module_##adapter##_init)
 
+#define DECLARE_MODULE_ADAPTER_STEP1(adapter, uuid, tr) \
+static struct comp_dev *module_##adapter##_shim_new(const struct comp_driver *drv, \
+					 const struct comp_ipc_config *config, \
+					 const void *spec) \
+{ \
+	return module_adapter_new(drv, config, &(adapter), spec);\
+} \
+\
+static const struct comp_driver comp_##adapter##_module = { \
+	.type = SOF_COMP_MODULE_ADAPTER, \
+	.uid = SOF_RT_UUID(uuid), \
+	.tctx = &(tr), \
+	.ops = { \
+		.create = module_##adapter##_shim_new, \
+		.prepare = module_adapter_prepare, \
+		.params = module_adapter_params, \
+		.copy = module_adapter_copy, \
+		.cmd = module_adapter_cmd, \
+		.trigger = module_adapter_trigger, \
+		.reset = module_adapter_reset, \
+		.free = module_adapter_free, \
+		.set_large_config = module_set_large_config,\
+		.get_large_config = module_get_large_config,\
+		.get_attribute = module_adapter_get_attribute,\
+	}, \
+}; \
+\
+static SHARED_DATA struct comp_driver_info comp_module_##adapter##_info = { \
+	.drv = &comp_##adapter##_module, \
+}; \
+\
+UT_STATIC void sys_comp_module_##adapter##_init(void); \
```

Comment:
this shouldn't be needed here: this macro should be used in C files, not in headers, typically close to the end of the file, so adding this declaration there won't help anyone

Reply:
More components may need changes to be able to load to testbench. Remove this then?

Reply:
I mean just this line - it's a function forward declaration. This is a part of a macro, the macro is used in .c files. So you end up having a function declaration and its definition that immediately follows it, so the declaration isn't needed there.